### PR TITLE
Feature: action list (and docker local dev config)

### PR DESCRIPTION
**Pull request by pbek**
This is the implementation of #111 (pbek/nextcloud-gpodder#1) and a local dev environment in docker, which is very handy.

Currently looks like:
image

PS: I to added myself as contributor, if that's something you want. See https://github.com/pbek/nextcloud-gpodder/commit/79cc084e849ce62d9cd8f63888e1bc36e39b43f2.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+```
+
+There also is a [logging web interface](http://localhost:8081/index.php/settings/admin/logging).
+
+## Tip
+
+In case something is broken try to reset the container:
+
+```bash
+docker-compose build; docker-compose down; docker volume prune -f
```
Comment (thrillfall) [inline docker/README.md:27]: This has global effect well beyond this project. I'd rather not put this  here as a "tip".

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+log in to your Nextcloud instance.
+
+## Check nextcloud.log
+
+For debugging, you can show the `nextcloud.log`:
+
+```bash
+make show-log
+```
+
```
Comment (thrillfall) [inline docker/README.md:19]: What is the command to execute the tests? I'd like that in the readme

**Comment by thrillfall** [comment]
Awesome! Thanks for the local dev environment!!

File: docker/README.md (hunk 1 above)
Comment (pbek) [reply docker/README.md:27]: True

File: docker/README.md (hunk 2 above)
Comment (pbek) [reply docker/README.md:19]: There is no npm in the docker image yet.

Comment (pbek) [reply docker/README.md:19]: It's just the nextcloud:25-apache image

Comment (thrillfall) [reply docker/README.md:19]: I was thinking about phpunit tests

Comment (pbek) [reply docker/README.md:19]: There is no composer neither... Of course you could tinker with the Dockerfile to add all those to the container.

**Comment by pbek** [comment]
I'm content with the state of this PR now. Features, see https://github.com/pbek/nextcloud-gpodder/issues/1

**Comment by pbek** [comment]
image

**Comment by thrillfall** [comment]
please do not merge main into your branch but rebase instead. The main history is free of upstream merge commits and i'd like to keep it that way

**Comment by pbek** [comment]
Hm, GitHub did that...

**Comment by pbek** [comment]
I didn't manage to do a rebase without rewriting all my commits to the current date... Best do that yourself. 😅

File: docker/README.md (hunk 2 above)
Comment (thrillfall) [reply docker/README.md:19]: I added a how to run test in the readme

File: lib/Core/EpisodeAction/EpisodeActionData.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		$this->podcastUrl = $podcastUrl;
+		$this->episodeUrl = $episodeUrl;
+		$this->action = $action;
+		$this->position = $position;
+		$this->started = $started;
+		$this->total = $total;
+	}
+
+	/**
+	 * @return string|null
```
Comment (thrillfall) [inline lib/Core/EpisodeAction/EpisodeActionData.php:36]: Please drop all the phpdoc (in all files of this PR) unless it specifies exceptions or array element type

File: lib/Core/EpisodeAction/EpisodeActionReader.php [hunk 4]
```diff
@@ -4,11 +4,61 @@
...
+use OCP\ICacheFactory;
 
 class EpisodeActionReader {
 	private array $requiredProperties = ['podcast', 'episode', 'action', 'timestamp'];
+    private EpisodeActionRepository $episodeActionRepository;
+    private ?ICache $cache = null;
+    private IClient $httpClient;
 
-	/**
+    public function __construct(
```
Comment (thrillfall) [inline lib/Core/EpisodeAction/EpisodeActionReader.php:21]: please create a separate class for the functionality you added here and compose it with the existing class

File: tests/Unit/Core/EpisodeAction/EpisodeActionReaderTest.php [hunk 5]
```diff
@@ -4,11 +4,31 @@
...
+
+    /** @var ICacheFactory */
+    private $iCacheFactory;
+
+    /** @var EpisodeActionRepository */
+    private $episodeActionRepository;
+
+    protected function setUp(): void {
+        parent::setUp();
+        $this->clientService = $this->createMock(IClientService::class);
```
Comment (thrillfall) [inline tests/Unit/Core/EpisodeAction/EpisodeActionReaderTest.php:25]: please add tests for new/changed code. mocking all dependencies without testing anything just makes the existing tests pass but does not tell us if things work as expected

File: lib/Core/EpisodeAction/EpisodeActionReader.php (hunk 4 above)
Comment (pbek) [reply lib/Core/EpisodeAction/EpisodeActionReader.php:21]: Might take a while until I rewrite all that. I first need to get a release of the action list on my prod server and then if I still have time I want to find out why I can't seem to be able to inject CSP settings on the admin page...

File: tests/Unit/Core/EpisodeAction/EpisodeActionReaderTest.php (hunk 5 above)
Comment (pbek) [reply tests/Unit/Core/EpisodeAction/EpisodeActionReaderTest.php:25]: Same as https://github.com/thrillfall/nextcloud-gpodder/pull/114#discussion_r1117890666

File: lib/Core/EpisodeAction/EpisodeActionReader.php (hunk 4 above)
Comment (pbek) [reply lib/Core/EpisodeAction/EpisodeActionReader.php:21]: I wasn't able to use CSP rules on admin pages, so I put together a full app page (together with lots of other stuff what would be out of scope for this a sync service) and put it in https://apps.nextcloud.com/apps/nextpod. The app depends on gpoddersync as sync service.

I'll close this PR for now.

**Pull request closed without merging**